### PR TITLE
Implement MySGonks page UI 

### PR DESCRIPTION
**Pull request by lin8231**
* Just UI, no data 
Looks like this
Screen Shot 2021-02-02 at 12 30 41 pm

Note that this branch is branched off fe-link-to-servlets
Only the last 4 commits displayed belong to this PR

### CODE REVIEW

File: project/frontend/src/App.js [hunk 1]
```diff
@@ -37,11 +40,62 @@ function App() {
...
+        fetch("./trending")
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
+              return {
+                ...prevState,
+                trending: data,
+              };
+            });
+          }),
```
Comment (texm) [inline project/frontend/src/App.js:91]: I feel like the logic in these fetch calls could be extracted into a function that returns a promise, for example
`fetchAndUpdateCompetitionInfo(apiEndpoint, stateKey)`

And also using ES6 [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

Would end up looking something like
```js
fetchAndUpdateCompetitionInfo(
    `./competitionInfo?user=${user.id}&competition=${compId}`, 
    "generalInfo")
```

Comment (lin8231) [reply project/frontend/src/App.js:91]: Just to clarify, which promise do you mean it should return at? The response promise or the jsonified promise?

Comment (texm) [reply project/frontend/src/App.js:91]: as in something like
```js
Promise.all([
  setLoading(true),
  fetchAndUpdateCompetitionInfo(`./competitionInfo?user=${user.id}&competition=${compId}, "generalInfo"),
  fetchAndUpdateCompetitionInfo("./trending", "trending"),
  // ... and so on
]).then(() => {
```

Comment (lin8231) [reply project/frontend/src/App.js]: Done.

Comment (lin8231) [reply project/frontend/src/App.js]: Please note that this was actually part of PR #95 (which I haven't merged) and this PR was only the last 4 commits before your comment (as indicated in the PR description) so the changes you requested were made in the branch of PR #95, and subsequently pulled from that branch so you can see them here as well - hence the merges

**Comment by roonil-wazlib** [comment]
@googlebot I consent.

**Review by texm — APPROVED** [review]

**Review by Phoebe-Khokgawe — APPROVED** [review]

**Comment by lin8231** [comment]
Changes are merged with #103 (which is a branch branched off this branch)

**Pull request closed without merging**